avocado/utils/cpu.py

```diff
@@ -152,6 +169,37 @@ def get_cpu_arch():
     return platform.machine()
 
 
+def get_x86_microarcitecture():
+    """
+    to get the x86 microarchiture like broadwell, Haswell
+    """
+    micro_arch_name = None
+    if get_cpu_vendor_name() == 'intel':
+        try:
+            with open('/sys/devices/cpu/caps/pmu_name', 'rb') as mico_arch:
+                micro_arch_name = mico_arch.read().decode('utf-8').strip('\n')
+        except Exception as err:
+            logging.warning("Could not find micro-architecture")
+    return micro_arch_name
+
+
+def get_powerpc_pvr():
+    """
+    to get powerpc pvr value
+    """
+    pvr = None
+    if get_cpu_arch() != 'powerpc':
+        return pvr
+    cpu_info = _get_cpu_info()
+    for item in cpu_info:
+        if b'revision' in item:
+            try:
+                pvr = float(re.findall(b'revision.*:\s+(\d+.\d+).*', item)[0])
```

Comment:
Looks like the pattern would only match the "revision" number, not the whole pvr hex code, would you please elaborate on your concern with this approach? thanks.
```
revision        : 1.0 (pvr 004e 0100)
```

Reply:
we do not use that hex code anywhere, all we need till now is revision no. + cpu version...
like power9 DD2.1 vs power9 DD2.2 etc... so we are good I believe, do you see any usecases ?

Reply:
@sathnaga if we get PVR number also that will be value add on